packages/swarm-docs/src/components/nav.js

```diff
@@ -3,21 +3,45 @@ import * as React from 'react';
 import { Link } from 'gatsby';
 
 type Props = {
-    fileNames: Array<string>
+    files: Array<Object>
+};
+
+const transformFiles = (files: Array<Object>) => {
+    const nav = [];
```

Comment:
not a huge deal, but building an array from each item of another array is what `map` is for - the current implementation has a fair amount of mutation, which can make it difficult to reason about and potentially more fragile.

```js
const transformFiles = (files: Array<Object>) =>
  files.map(file =>
    file.nodes.reduce((folderMap, node) => {
      // initialize directory key if it doesn't exist;
      if (!folderMap[node.relativeDirectory]) {
        folderMap[node.relativeDirectory] = [];
      }
      folderMap[node.relativeDirectory].push(node.name);
      return folderMap;
    }, {})
  );
```

Reply:
yeah, this is cleaner. 

on the note of `fair amount of mutation`, i looked into a bunch of queries for this and it _seems_ this is the best way to achieve this. i'm going to pair with @hsbacot on it and maybe we can find a better way.

Reply:
ah, the mutation I'm referring to is just mutating the `nav` array after it's instantiated, and the `folderMap` object - using `map` and `reduce` allows those operations to be 'pure' inside the callback rather than interacting with variables in parent scope